wfcGen.py

```diff
@@ -0,0 +1,287 @@
+        else:
+            self.entropy = [self.id]
+        
+    def getEntropy(self):
+        if self.collapsed:
+            return -1
+        else:
+            if self.anyFlag:
+                return len(cells.keys())
+            else:
+                return len(self.entropy)
+
+class Grid:
+    def __init__(self, sx, sy):
+        self.sx = sx
+        self.sy = sy
+        self.entropySave = {}
+        self.grid = []
+        self.last5 = []
+        self.GIMAGE = None
+        self.backUps = []
+        self.backUpsLength = 20
+        self.reloads = 0
+    
+    def setRandom(self):
+        self.grid = []
+        for y in range(self.sy):
+            self.grid.append([])
+            for x in range(self.sx):
+                self.grid[y].append(Cell([x, y]))
+        self.grid[random.randint(0, self.sy-1)][random.randint(0, self.sx-1)].setCell(random.randint(0, len(cells)-1))
+        self.getFinalImage()
+    
+    def loadEntropy(self):
+        self.entropySave = {}
+        for y in range(0, self.sy):
+            for x in range(0, self.sx):
+                if not self.grid[y][x].collapsed:
+                    self.grid[y][x].loadEntropy(self.grid)
+                    if self.grid[y][x].getEntropy() in self.entropySave.keys():
+                        self.entropySave[self.grid[y][x].getEntropy()].append([x, y])
+                    else:
+                        self.entropySave[self.grid[y][x].getEntropy()] = [[x, y]]
+    
+    def collapse(self):
+        try:
+            self.loadEntropy()
+            s = sorted(self.entropySave.keys())
+            while min(s) < 1:
+                s.remove(min(s))
+            y = self.entropySave[s[0]][0][1]
+            x = self.entropySave[s[0]][0][0]
+            
+            self.grid[y][x].collapse()
+            self.loadEntropy()
+            flag1 = 0 in self.entropySave.keys()
+            while 0 in self.entropySave.keys():
+                if len(self.backUps) > 0:
+                    lb = self.backUps[len(self.backUps)-1]
+                    self.grid[lb[1]][lb[0]] = Cell(lb)
+                    self.backUps = self.backUps[:-1]
+                else:
+                    self.setRandom()
+                self.loadEntropy()
+            if flag1:
+                self.reloads += 2
+                for j in range(self.reloads):
+                    if len(self.backUps) > 0:
+                        lb = self.backUps[len(self.backUps)-1]
+                        self.grid[lb[1]][lb[0]] = Cell(lb)
+                        self.backUps = self.backUps[:len(self.backUps)-1]
+                    else:
+                        self.setRandom()
+            else:
+                self.reloads = 0
+            self.backUps.append([x, y])
+
+            if ITERATION % 400 == 0:
+                self.getFinalImage()
+            return False
+        except:
+            return True
+        
+    
+    def getFinalImage(self):
+        if self.GIMAGE == None:

```

Comment:
In python the `is` operator is commonly used here:

```suggestion
        if self.GIMAGE is None:
```

Technical reason: https://stackoverflow.com/questions/14247373/python-none-comparison-should-i-use-is-or

I don't think it often happens that it makes a difference but... ¯\_(ツ)_/¯